api/v1alpha1/utils/commons.go

```diff
@@ -74,6 +81,11 @@ type VaultConnection struct {
 	MaxRetries *int `json:"maxRetries,omitempty"`
 }
 
+type VaultClientCache struct {
+	clients map[string]*vault.Client
```

Comment:
let's use a sync.Map[1] here so we don't have to worry about the mutex
[1]: https://pkg.go.dev/sync#Map

Reply:
Thank you, done